services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3263,16 +3263,19 @@ private void deleteOneStoreVersion(String clusterName, String storeName, int ver
       deleteHelixResource(clusterName, resourceName);
       LOGGER.info("Killing offline push for: {} in cluster: {}", resourceName, clusterName);
       killOfflinePush(clusterName, resourceName, true);
+
+      // Check DIV error in the push status before stopping the monitor.
+      boolean hasFatalDataValidationError = hasFatalDataValidationError(pushMonitor, resourceName);
```

Comment:
Looks like the resources on which the check is made is different across two code path? Is that intended?
The description only calls about moving out the place where this check happens but not necessarily changing the resource itself. e.g., Old one uses `versionTopicName` while this one uses `resourceName`. 

Are both same?

Reply:
Yeah, good point and I also noticed that. Thus, I took a look at `deleteVersionFromStoreRepository` then realized that returned `deletedVersion` is Optional whose version number is either `versionNumber` or `empty`. 

Then, if consider that when we used the returned boolean `hasFatalDataValidationError` , it already passed the condition `deletedVersion.isPresent()`, then both value have to be the same.